src/sentry/grouping/ingest/hashing.py

```diff
@@ -213,17 +213,20 @@ def get_or_create_grouphashes(
     event: Event,
     project: Project,
     variants: dict[str, BaseVariant],
-    hashes: Sequence[str],
+    hashes: Iterable[str],
     grouping_config: str,
 ) -> list[GroupHash]:
     is_secondary = grouping_config == project.get_option("sentry:secondary_grouping_config")
     grouphashes: list[GroupHash] = []
 
-    # The only utility of secondary hashes is to link new primary hashes to an existing group.
-    # Secondary hashes which are also new are therefore of no value, so there's no need to store or
-    # annotate them and we can bail now.
-    if is_secondary and not GroupHash.objects.filter(project=project, hash__in=hashes).exists():
-        return grouphashes
+    if is_secondary:
+        # The only utility of secondary hashes is to link new primary hashes to an existing group
+        # via an existing grouphash. Secondary hashes which are new are therefore of no value, so
+        # filter them out before creating grouphash records.
+        hashes = filter(
+            lambda hash_value: GroupHash.objects.filter(project=project, hash=hash_value).exists(),
+            hashes,
+        )
```

Comment:
if i'm reading this correctly, we'll make a database query per hash. could we instead do something like:
```suggestion
    existing_hashes = set(GroupHash.objects.filter(
        project=project,
        hash__in=hashes
    ).values_list('hash', flat=True))
    filtered_hashes = [hash_value for hash_value in hashes if hash_value in existing_hashes]
```

Reply:
We could, but given that there's max 2 hashes we're talking about here... maybe over optimization? WDYT?

Reply:
if it weren't in ingestion i'd say definitely not. but it is marginally worth it IMO. 

to also protect against any future changes in the system (what if 5 years from now someone decides we need to compute 10 secondary hashes) etc. etc. far-fetched, but it doesn't hurt.

Reply:
Okay, I updated it.